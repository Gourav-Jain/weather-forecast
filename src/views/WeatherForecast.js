import React from 'react';
import { useWeatherForecastContainer } from './WeatherForecast.container'
import DayCard from '../components/DayCard'

export const WeatherForecast = () => {
  const { error, cityData, forecastData } = useWeatherForecastContainer();

  let element = <h1 style={{color:"red"}}>{error}</h1> 
  if (!error && (forecastData && forecastData.length > 0)) {
    element = <div>
      <h1>{`${cityData.city}, ${cityData.country}`}</h1>
      <DayCard forecastData={forecastData} />
    </div>
  }
  
  return (
    <>
      { element }
    </>
  )
}

export default WeatherForecast;