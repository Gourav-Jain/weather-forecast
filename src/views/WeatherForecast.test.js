import React from 'react';
import { render } from '@testing-library/react';
import WeatherForecast from './WeatherForecast';

test('@WeatherForecast', () => {
  const { container } = render(<WeatherForecast />);
  expect(container).toMatchSnapshot()
});