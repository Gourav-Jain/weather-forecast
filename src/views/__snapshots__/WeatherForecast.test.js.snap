// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@WeatherForecast 1`] = `
<div>
  <h1
    style="color: red;"
  />
</div>
`;
